### PR TITLE
Use onnx-to-krnl and krnl-to-llvm to handle NNPA lowerings

### PR DESCRIPTION
**Pull request by tungld**
Currently there is code duplication between onnx-to-krnl and zhigh-to-zlow, and between krnl-to-llvm and zlow-to-llvm, which is not good and error-prone.

This patch removes the duplication by just using onnx-to-krnl and krnl-to-llvm and providing hooks for accelerators. This mechanism is consistent with what we are doing to support accelerators in `onnx-mlir` and `onnx-mlir-opt` drivers.

In summary, to support a new accelerator, users are expected to write hooks for 4 components: `onnx-mlir`, `onnx-mlir-opt`, `onnx-to-krnl`, and `krnl-to-llvm` in their own accelerator class derived from the Accelarator class.

### CODE REVIEW

File: src/Accelerators/NNPA/Compiler/NNPACompilerUtils.cpp [hunk 1]
```diff
@@ -143,9 +111,8 @@ void addPassesNNPA(mlir::OwningOpRef<mlir::ModuleOp> &module,
       emissionTarget = EmitMLIR;
     else {
       pm.addPass(mlir::createCanonicalizerPass());
-      // Add instrumentation for remaining Onnx Ops
-      if (instrumentZHighOps != "" && instrumentZHighOps != "NONE")
-        pm.addNestedPass<FuncOp>(createInstrumentONNXPass());
+      // Add instrumentation for ZHigh Ops
+      pm.addNestedPass<FuncOp>(onnx_mlir::zhigh::createInstrumentZHighPass());
```
Comment (etiotto) [inline src/Accelerators/NNPA/Compiler/NNPACompilerUtils.cpp:115]: `onnx_mlir::`   shouldn't be required given we are inside that namespace already.

File: src/Accelerators/NNPA/Conversion/ZLowToLLVM/ZLowToLLVM.hpp [hunk 2]
```diff
@@ -0,0 +1,26 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+//===---------- ZLowToLLVM.cpp - Lowering from ZLow to LLVM ---------------===//
```
Comment (etiotto) [inline src/Accelerators/NNPA/Conversion/ZLowToLLVM/ZLowToLLVM.hpp:5]: ZLowToLLVM.cpp -> ZLowToLLVM.hpp

File: src/Accelerators/NNPA/Compiler/NNPACompilerUtils.cpp (hunk 1 above)
Comment (tungld) [reply src/Accelerators/NNPA/Compiler/NNPACompilerUtils.cpp]: Done. Thanks!

File: src/Accelerators/NNPA/Conversion/ZLowToLLVM/ZLowToLLVM.hpp (hunk 2 above)
Comment (tungld) [reply src/Accelerators/NNPA/Conversion/ZLowToLLVM/ZLowToLLVM.hpp]: Done. Thanks!

**Review by etiotto — APPROVED** [review]
LGTM

**Comment by tungld** [comment]
@jenkins-droid test this please

**Comment by tungld** [comment]
@jenkins-droid test this please

**Pull request merged**